processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -462,7 +462,7 @@ private Assignment assignToCollection(Type targetType,
                     // target accessor is setter, so wrap the setter in setter map/ collection handling
                     result = new SetterWrapperForCollectionsAndMaps(
                         result,
-                        Executables.getCollectionGetterName( targetWriteAccessor ),
+                        targetReadAccessor.getSimpleName().toString(),
```

Comment:
Didn't we have a problem with the toString() method in different Compiler-Implementations? Well, all tests pass, so it seems to be okay. :grin: